File: Leetcode/Easy/121. Best Time to Buy and Sell Stock/solution.js
/**
 * @param {number[]} prices
 * @return {number}
 */

 var maxProfit = function(prices) {
    //max profit
    let p = 0;
    
    for (let b = 0, s = 1; s < prices.length; s++) {
        if (prices[b] < prices[s]) {
            if (prices[s] - prices[b] > p) {
                p = prices[s] - prices[b];
            }
        }
        
        else if (prices[b] >= prices[s]) {
            b = s;
        }
    }
    return p;
};

File: Leetcode/Easy/121. Best Time to Buy and Sell Stock/solution.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        # left right
        # if right is lower than left, new left it is
        # if right is larger calculate max profit

        maxProfit = 0
        left = 0
        right = 1

        while (right < len(prices)):
            if prices[right] > prices[left]:
                profit = prices[right]- prices[left]
                if profit > maxProfit:
                    maxProfit = profit
            elif prices[right] < prices[left]:
                left = right
            
            right += 1
        
        return maxProfit